src/org/thoughtcrime/securesms/ConversationUpdateItem.java

```diff
@@ -80,6 +88,22 @@ public DcMsg getMessageRecord() {
   }
 
   private void setGenericInfoRecord(DcMsg messageRecord) {
+    DcMsg parentMsg = messageRecord.getParent();
```

Comment:
This does two database requests: one for finding the parent, one for loading it from the database -> probably making scrolling less smooth.

Maybe we should cache the parent message in DcMsg, or do the loading asynchronously, or both.

Reply:
we can avoid the additional request by checking for `dc_msg_get_info_type(infoMsg) == DC_INFO_WEBXDC_INFO_MESSAGE` - and only if that condition is true, load the parent.

this is also what ios and probably also desktop are doing.

Reply:
> we can avoid the additional database request by checking for [`dc_msg_get_info_type() == DC_INFO_WEBXDC_INFO_MESSAGE`](https://c.delta.chat/classdc__msg__t.html#ab3eeb59143e67fda31d3648bd775f77b)- and only if that condition is true, load the parent.

done :heavy_check_mark: